samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/AssessmentService.java

```diff
@@ -1118,6 +1118,191 @@ public SecurityAdvice isAllowed(String arg0, String arg1,
+					while (iterAS.hasNext()) {
+						AnswerIfc answer = iterAS.next();
+						
+						markupText.append("\n");
+						
+    					if (answer.getIsCorrect()) {
+    						markupText.append("*");
+    					}
+    					if (TypeIfc.MULTIPLE_CHOICE.intValue() == item.getTypeId() 
+    							|| TypeIfc.MULTIPLE_CORRECT.intValue() == item.getTypeId()) {
+    						markupText.append(answer.getLabel() + ". ");
+    					}
+    					
+    					if (TypeIfc.FILL_IN_NUMERIC.intValue() == item.getTypeId()) {
+    						markupText.append("{" + answer.getText() + "}");
+    					}
+    					else if (TypeIfc.TRUE_FALSE.intValue() == item.getTypeId()) {
+    						String boolText = bundle.get("false");
+    						if (Boolean.parseBoolean(answer.getText())) {
+    							boolText = bundle.get("true");
+    						}
+    						markupText.append(boolText);
+    					}
+    					else {
+    						markupText.append(answer.getText());
+    					}
+					}
+				}
+				
+				String randomized = item.getItemMetaDataByLabel(ItemMetaDataIfc.RANDOMIZE);
+				if (randomized != null && Boolean.valueOf(randomized)) {
+					markupText.append("\n");
+					markupText.append(bundle.get("randomize"));
+				}
+				
+				if (item.getHasRationale() != null && item.getHasRationale()) {
+					markupText.append("\n");
+					markupText.append(bundle.get("rationale"));
+				}
+				
+				if (item.getCorrectItemFeedback() != null && !"".equals(item.getCorrectItemFeedback())) {
+					markupText.append("\n");
+					markupText.append("#FBOK:" + item.getCorrectItemFeedback());
+				}
+				
+				if (item.getInCorrectItemFeedback() != null && !"".equals(item.getInCorrectItemFeedback())) {
+					markupText.append("\n");
+					markupText.append("#FBNOK:" + item.getInCorrectItemFeedback());
+				}
+				markupText.append("\n");
+				
+				nQuestion++;
+			}
+		}
+		
+		return markupText.toString();
+	}	  
+    
+	/**
+	 * Check if there are questions not exportable to markup text
+	 * 
+	 * @param assessment
+	 * @return
+	 */
+	public boolean isExportable(AssessmentFacade assessment) {
+		boolean exportToMarkupText = false;
+		
+		Iterator<SectionFacade> sectionIter = assessment.getSectionArray().iterator();
+		while (sectionIter.hasNext() && !exportToMarkupText) {
+			SectionFacade section = sectionIter.next();
+			List<ItemDataIfc> items;
+						
+			if (section.getSectionMetaDataByLabel(SectionDataIfc.AUTHOR_TYPE).equals(SectionDataIfc.QUESTIONS_AUTHORED_ONE_BY_ONE.toString()))
+  			{
+				items = section.getItemArray();
+  			}
+			else 
+			{
+				QuestionPoolService qpService = new QuestionPoolService();
+				items = qpService.getAllItems(Long.valueOf(section.getSectionMetaDataByLabel(SectionDataIfc.POOLID_FOR_RANDOM_DRAW)));
+			}
+  				
+			for (ItemDataIfc item : items) 
+			{
+				// only exports these questions types
+				if (TypeIfc.MULTIPLE_CHOICE.intValue() != item.getTypeId() && TypeIfc.MULTIPLE_CORRECT.intValue() != item.getTypeId() 
```

Comment:
I think I've seen this long `if` statement previously in this patch, might be a good opportunity to extract into a method to improve readability and reduce code duplication.